setup.py

```diff
@@ -44,6 +44,7 @@ def read_version():
     "packaging>=20.0",
     "pandas",
     "pathos",
+    "semantic-version",
```

Comment:
do we need to add extra package?  Whats the use of this?

Reply:
There are reservations to add new dependencies in SDK due to some inherrant constraints. Is there a possibility to use some other alternatives currently existing in SDK for versioning? Can we use `packaging.version` [here](https://github.com/aws/sagemaker-python-sdk/blob/dev/setup.py#L44).

It seems like after adding the dependency the size increased by +17KB
```
-rw-r--r--   1 mufi  staff   449K Dec  6 12:51 sagemaker-2.69.1.dev0.tar.gz
-rw-r--r--   1 mufi  staff   466K Dec  6 12:55 sagemaker-2.71.1.dev0.tar.gz
```